pkg/cmd/printer/printer_parquet.go

```diff
@@ -0,0 +1,176 @@
+	ThreadStartTime     int                 `parquet:"name=threadStartTime, type=INT64"`
+	ProcessorID         int                 `parquet:"name=processorId, type=INT64"`
+	ProcessID           int                 `parquet:"name=processId, type=INT64"`
+	CgroupID            int                 `parquet:"name=cgroupId, type=INT64, convertedtype=UINT_64"`
+	ThreadID            int                 `parquet:"name=threadId, type=INT64"`
+	ParentProcessID     int                 `parquet:"name=parentProcessId, type=INT64"`
+	HostProcessID       int                 `parquet:"name=hostProcessId, type=INT64"`
+	HostThreadID        int                 `parquet:"name=hostThreadId, type=INT64"`
+	HostParentProcessID int                 `parquet:"name=hostParentProcessId, type=INT64"`
+	UserID              int                 `parquet:"name=UserID, type=INT64"`
+	MountNS             int                 `parquet:"name=mountNamespace, type=INT64"`
+	PIDNS               int                 `parquet:"name=pidNamespace, type=INT64"`
+	ProcessName         string              `parquet:"name=processName, type=BYTE_ARRAY"`
+	Executable          ParquetFile         `parquet:"name=executable"`
+	HostName            string              `parquet:"name=hostName, type=BYTE_ARRAY, convertedtype=UTF8"`
+	ContainerID         string              `parquet:"name=containerId, type=BYTE_ARRAY, convertedtype=UTF8"`
+	Container           ParquetContainer    `parquet:"name=container"`
+	Kubernetes          ParquetKubernetes   `parquet:"name=kubernetes"`
+	EventID             int                 `parquet:"name=eventId, type=INT64"`
+	EventName           string              `parquet:"name=eventName, type=BYTE_ARRAY, convertedtype=UTF8"`
+	MatchedPolicies     []string            `parquet:"name=matchedPolicies, type=MAP, convertedtype=LIST, valuetype=BYTE_ARRAY, valueconvertedtype=UTF8"`
+	ArgsNum             int                 `parquet:"name=argsNum, type=INT64"`
+	ReturnValue         int                 `parquet:"name=returnValue, type=INT64"`
+	Syscall             string              `parquet:"name=syscall, type=BYTE_ARRAY, convertedtype=UTF8"`
+	StackAddresses      []int64             `parquet:"name=stackAddresses, type=MAP, convertedtype=LIST, valuetype=INT64, valueconvertedtype=UNIT_64"`
+	ContextFlags        ParquetContextFlags `parquet:"name=contextFlags"`
+	ThreadEntityId      int32               `parquet:"name=threadEntityId, type=INT32, convertedtype=UINT_32"`
+	ProcessEntityId     int32               `parquet:"name=processEntityId, type=INT32, convertedtype=UINT_32"`
+	ParentEntityId      int32               `parquet:"name=parentEntityId, type=INT32, convertedtype=UINT_32"`
+	Args                string              `parquet:"name=args, type=BYTE_ARRAY, convertedtype=JSON"`
+}
+
+/*
+The following structs are not necessary: ParquetContainer, ParquetKubernetes, ParquetContextFlags, ParquetFile.
+
```

Comment:
Yep, check my comment for the PR. I think we should try to have only the "really needed types", serialize with a custom marshaller whatever is needed and can't be automatically type-converted, and go from there (at least until we have protobufs fully done, having the event structure way more stable than it is now (soon).

Reply:
I couldn't add the parquet tags to existing types (all the inner structs), since they are on a different package:
github.com/aquasecurity/tracee/types v0.0.0-20231013014739-b32a168ee6a8

Since the Event struct contains types which are not supported by parquet, I can't use it.
For example I can add 
`CgroupID uint json:"cgroupId" parquet:"name=cgroupId, type=INT64, convertedtype=UINT_64"`
But I get `error reflect: call of reflect.Value.Int on uint32 Value`

And I couldn't find a way to implement a custom marshal function.
The marshal function doesn't support it:
https://github.com/xitongsys/parquet-go/blob/8ca067b2bd324788a77bf61d4e1ef9a5f8b4b1d2/marshal/marshal.go#L293C16-L293C16

It calls directly to the ParquetStruct Marshal function

Reply:
Ok so here is the thing, I played around a bit with parquet, because it was new to me and I wanted to understand better how it worked. 

It ended up being good and bad. Good that I played and bad related to how this `logical types` of parquet cast unsigned values (and also due to the issue https://github.com/aquasecurity/tracee/issues/3690 just discovered).

I was going to suggest you using json schema to parquet (https://gist.github.com/rafaeldtinoco/fbe58a020fd0d73ea37ce6980520f6ea) so we didnt have to maintain type duplicates. You would complain back to me saying tracee event type -> json -> parquet could add overhead, I would request you to check if that was too bad.

During this process, I discovered https://github.com/aquasecurity/tracee/issues/3690. It basically means that many other fields from the tracee event struct should be unsigned integers (I would say from 4 to 8 fields should have their types changed). Not sure how we missed that, but during the decode phase (from eBPF to our internal types) we converted them to Ints and never checked for overflows.

Good thing that parquet does not support unsigned values (actually it's a terrible thing :P) so I detected the overflow when generating the parquet file, I believe.

Ok, with all that said, and thinking about the near future, we're moving away from these internal types to the new protobuf types you will find in `github.com/aquasec/tracee/api` (specially in `event.pb`.

The protobuf types are declared correctly (uint32s, uints64) but it might impose a very hard time for you IF you rely in a parallel Parquet type to each type automatically generated from the protobuf (and actually I don't think we want parallel types just for marshalling logic).

That explains why I was going after the JSON Schema (to give you an alternative). Why ? 

1. I believe it might be easier for you to convert types from json Integers (than to worry about golang types coming from internal structures) to parquet types.

2. Its likely that tracee will eventually move into a grpc model where the engine will be a grpc server providing streams. Those streams will come serialized with protobuf types but we could also have an already marshalled version. You can see a very simple example at https://github.com/josedonizetti/tracee-grpc-client/blob/main/examples/stream_events/main.go for now.

3. Of course you can work through parquet marshalling and our protobuf types, I just dont want to commit to accept your printer now (when we fix the wrong field types), having duplicate types, and then discover your printer wont work with our near future protobuf work (and either remove your printer or request you to change and be blocked until you do).

Does it make sense for you ?

Suggestion of course of action:

- We fix the event struct types
- you rely in json schema for the printer (similar to my gist)
- you investigate the protobuf types we have for further actions
- by relying in the json schema you will probably remove headaches from protobuf (and we could have a stream of json marshaled types when grpc streams are finished). 

With all that, Im trying to give you a "roadmap" if you stick to tracee as your "engine" (for the project of yours). 

I hope it helps. No hard commitments on our side, but for sure a best effort trying to help.